src/dehydrated_devices.rs

```diff
@@ -33,18 +71,53 @@ impl DehydratedDevices {
     #[wasm_bindgen]
     pub async fn rehydrate(
         &self,
-        pickle_key: &Uint8Array,
+        dehydrated_device_key: &DehydratedDeviceKey,
         device_id: &DeviceId,
         device_data: &str,
     ) -> Result<RehydratedDevice, JsError> {
-        let pickle_key: [u8; 32] =
-            pickle_key.to_vec().try_into().map_err(|_| JsError::new("Wrong key length"))?;
         Ok(self
             .inner
-            .rehydrate(&pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
+            .rehydrate(
+                &dehydrated_device_key.inner,
+                &device_id.inner,
+                serde_json::from_str(device_data)?,
+            )
             .await?
             .into())
     }
+
+    /// Get the cached dehydrated device key if any.
+    ///
+    /// `None` if the key was not previously cached (via
+    /// [`DehydratedDevices::save_dehydrated_device_key`]).
```

Comment:
oh, another thing: this style of link doesn't work for the wasm bindings. You have to use `{@link ...}`, as we do for typescript.

Unfortunately we haven't wired up generating the documentation in CI. Could you generate it locally (`yarn doc`) and check the results?

Reply:
Thx fixed this one and another in this file